src/components/Quotes.js

```diff
@@ -0,0 +1,31 @@
+import React, { useState, useEffect } from 'react';
+
+export default function Quotes() {
+  const [quote, setQuote] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [boolError, setBoolError] = useState(false);
+
+  const fetchQuote = async () => {
+    setIsLoading(true);
+    try {
+      const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=cool',
+        {
+          method: 'GET',
+          headers: { 'X-Api-Key': '8SSMZWYI21/6Rj0KXlch7A==FiOOQrXcbRQ0WIXK' },
+        });
+      const json = await response.json();
+      setQuote(json[0].quote);
+    } catch (error) {
+      setBoolError(true);
+    }
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    fetchQuote();
+  }, []);
```

Comment:
- [ ] The quotes seem to be re-fetching in one view and for some time it gives an error. You could move the fetching method into the useEffect and then pass it into the dependency array, the external variables used, setBoolError, and setIsLoading. Right now, the useEffect is causing a memory leak which is why the quotes are being re-fetched and re-rendering. Kindly work on that so the useEffect hook does not cause a memory leak which is causing re-fetching and re-rendering without refreshing the browser. 👍🏽

Reply:
@AmaduKamara Hi, yes it fetches twice because in development mode useEffect is called twice, that is how React works. The only way to make useEffect run once in development is to remove strict mode which will trigger linters.

Now since useEffect is called twice, these are the expected combination of results:
1st run      2nd run
Error         Error
Quote       Quote
Error         Quote
Quote       Error

The API server CORS policy is not properly set up for localhost:3000, so it's a hit or a miss. This is the reason for the ambiguous results. I have cloned a few of my peers' repo and verified these findings.

I implemented the changes you suggested, useeffect still run twice (it's a react feature) and the API call response is still ambiguous. I will commit and push these changes and ask for a review again.

Reply:
Hi @Anand8317,
Well, I didn't know that about the API but it's good you brought that up.
I understand how useEffect works on development due to the strict mode. My thought about the error was the way you used the function inside the useEffect. I modified your code and placed the fetch function inside the useEffect and then pass the setter functions into the useEffect dependency array, then things work fine as expected, one time it shows an error, and another load time, it shows a quote, and another load time it shows loading which I'm sure due to the API setup as you said. But it's ok you did the changes.

Reply:
Highly appreciate the help. Thank you.